python/pyADflow.py

```diff
@@ -1309,6 +1309,13 @@ def evalFunctionsSens(self, aeroProblem, funcsSens, evalFuncs=None):
 
         self.setAeroProblem(aeroProblem)
 
+        # we now need to reset the adjoint fail flag. This may be the
+        # first time we are setting this for this AP, or we may have
+        # a fail flag remaining from a previous call. Either way, we
+        # want to at least try solving the adjoints here, regardless
+        # of what happened in the previous iterations.
+        self.curAP.adjointFailed = False
```

Comment:
yeah I know that, but my idea was to just keep that for the cases where `solveadjoint` is called from some other routine.

Reply:
maybe we should remove that alltogether and have it throw an attribute error, since you should not go into that part of the code without the attribute set properly anyways.